SCClassLibrary/Common/GUI/PlusGUI/Control/ServerPlusGUI.sc

```diff
@@ -360,166 +360,29 @@
-							ytabs * tabSize + 0.5,
-							parent.bounds.width - (xtabs * tabSize * 2),
-							thisSize * tabSize;
-						);
-						pen.fillColor = Color.grey(0.8);
-						pen.fillRect(rect);
-						pen.strokeRect(rect);
-						pen.color = Color.black;
-						pen.stringInRect(
-							" Group" + node.key.asString +
-							(node.key == 1).if("- default group", ""),
-							rect
-						);
-						drawFunc.value(node.value);
-						ytabs = endYTabs;
-					},{
-						rect = Rect(xtabs * tabSize + 0.5,
-							ytabs * tabSize + 0.5,
-							7 * tabSize,
-							0.8 * tabSize
-						);
-						pen.fillColor = Color.white;
-						pen.fillRect(rect);
-						pen.strokeRect(rect);
-						pen.color = Color.black;
-						pen.stringInRect(
-							" " ++ node.key.asString + node.value.asString,
-							rect
-						);
-						ytabs = ytabs + 1;
-					});
-				});
-				xtabs = xtabs - 1;
-			};
-			drawFunc.value(levels);
-		};
-
-		// msg[1] controls included
-		// msg[2] nodeID of queried group
-		// initial number of children
-		resp = OSCFunc({ arg msg;
-			var finalEvent;
-			var i = 2, j, controls, printControls = false, dumpFunc;
-			if(msg[1] != 0, {printControls = true});
-			dumpFunc = {|numChildren|
-				var event, children;
-				event = ().group;
-				event.id = msg[i];
-				event.instrument = nil; // need to know it's a group
-				i = i + 2;
-				children = Array.fill(numChildren, {
-					var id, child;
-					// i = id
-					// i + 1 = numChildren
-					// i + 2 = def (if synth)
-					id = msg[i];
-					if(msg[i+1] >=0, {
-						child = dumpFunc.value(msg[i+1]);
-					}, {
-						j = 4;
-						child = ().synth.instrument_(msg[i+2]);
-						if(printControls, {
-							controls = ();
-							msg[i+3].do({
-								controls[msg[i + j]] = msg[i + j + 1];
-								j = j + 2;
-							});
-							child.controls = controls;
-							i = i + 4 + (2 * controls.size);
-						}, {i = i + 3 });
-					});
-					child.id = id;
-				});
-				event.children = children;
-				event;
-			};
-			finalEvent = dumpFunc.value(msg[3]);
-			done = true;
-			collectChildren = {|group|
-				group.children.collect({|child|
-					if(child.children.notNil,{
-						child.id -> collectChildren.value(child);
-					}, {
-						child.id -> child.instrument;
-					});
-				});
-			};
-			levels = collectChildren.value(finalEvent);
-			countSize = {|array|
-				var size = 0;
-				array.do({|elem|
-					if(elem.value.isArray, { size = size + countSize.value(elem.value) + 2}, {size = size + 1;});
-				});
-				size
-			};
-			defer {
-				view.bounds = Rect(0, 0, 400, max(400, tabSize * (countSize.value(levels) + 2)));
-				view.refresh;
-			}
-		}, '/g_queryTree.reply', addr).fix;
-
-		updateFunc = {
-			updater = fork {
-				loop {
-					this.sendMsg("/g_queryTree", 0, 0);
-					interval.wait;
-				}
+	plotTree { |interval=0.5, bounds|
+		var detectWidth = { |thisRecBounds|
+			if(thisRecBounds.asRect.width < 400) {
+				var bnds = thisRecBounds.asRect;
```

Comment:
here, there are similar calculations as needed in the other merge request. Maybe you can unify them?